src/pages/SignUpForm.js

```diff
@@ -1,51 +1,57 @@
-import React, { useState, useEffect } from 'react';
-import { useDispatch } from 'react-redux';
-import { useNavigate, Link } from 'react-router-dom';
-import { signup } from '../redux/user/userAPI';
+import React, { useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
 import 'bootstrap/dist/css/bootstrap.min.css';
 import '../styles/common.css';
 
 const SignUp = () => {
-  const dispatch = useDispatch();
   const navigate = useNavigate();
 
-  const [value, setValue] = useState({
+  const [user, setUser] = useState({
     name: '',
     email: '',
     password: '',
-    confirmPassword: '',
+    password_confirmation: '',
+    avatar: null,
   });
 
-  const [passwordMatchError, setPasswordMatchError] = useState(false);
-
-  const onChange = (e) => {
-    setValue({
-      ...value,
-      [e.target.id]: e.target.value,
+  const handleChange = (event) => {
+    setUser({
+      ...user,
+      [event.target.name]: event.target.value,
     });
   };
 
-  const createUser = (e) => {
-    e.preventDefault();
-    if (value.password === value.confirmPassword) {
-      dispatch(signup(value, navigate));
-      setPasswordMatchError(false);
-    } else {
-      setPasswordMatchError(true);
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+    try {
+      const response = await fetch('http://127.0.0.1:4000/api/v1/auth/sign_up', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ user }),
+      });
+
+      if (response.ok) {
+        await response.json();
+        navigate('/signin');
+      } else {
+        const error = await response.json();
+        console.error(error);
+      }
+    } catch (error) {
+      console.error(error);
```

Comment:
Handle errors in a more efficient way. So that users can become aware of it.  the console log is not a good option.